### PR TITLE
[mdatagen] Revert revert of mdatagen templateFS

### PR DESCRIPTION
**Pull request by MovieStoreGuy**
**Description:** <Describe what has changed.>

**Link to tracking Issue:** 
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/17576

**Testing:** 
Already covered, including windows tests.

**Documentation:** 
NA

### CODE REVIEW

**Comment by MovieStoreGuy** [comment]
I unfortunately don't have a windows CI platform to test on personally, so excuse me while I abuse the CI system

File: cmd/mdatagen/main.go [hunk 1]
```diff
@@ -104,7 +99,7 @@ func generateFile(tmplFile string, outputFile string, md metadata) error {
 				},
 				"stringsJoin": strings.Join,
 				"inc":         func(i int) int { return i + 1 },
-			}).ParseFiles(tmplFile))
+			}).ParseFS(templateFS, strings.ReplaceAll(tmplFile, "\\", "/")))
```
Comment (MovieStoreGuy) [inline cmd/mdatagen/main.go:102]: So looking at the coder, it looks like the `\` is a special character used for `template.Glob` and caused issues there.

Comment (codeboten) [reply cmd/mdatagen/main.go:102]: Please add a comment reflecting the reason for this `ReplaceAll` here.  Is this a known issue w/ ParseFS?

Comment (MovieStoreGuy) [reply cmd/mdatagen/main.go:106]: Added.

I haven't checked if there is a known issue with `ParseFS` for windows

Comment (MovieStoreGuy) [reply cmd/mdatagen/main.go:106]: From what I understand, it looks like it extends from https://pkg.go.dev/io/fs#ValidPath which forces `/` as the path separator regardless of OS.

**Review by dmitryax — APPROVED** [review]

**Pull request merged**